Classes/Command/NodeIndexCommandController.php

```diff
@@ -234,131 +196,377 @@ public function indexNodeCommand(string $identifier, string $workspace = null):
+     * @return int
+     * @throws Exception
+     */
+    protected function executeBuildWorkspaceCommand(array $arguments)
+    {
+        ob_start(null, 1<<20);
+        $commandIdentifier = 'flowpack.elasticsearch.contentrepositoryadaptor:nodeindex:buildworkspaceinternal';
+        $status = Scripts::executeCommand($commandIdentifier, $this->flowSettings, true, $arguments);
+        if ($status !== true) {
+            throw new Exception(\vsprintf('Command: %s with parameters: %s', [$commandIdentifier, \json_encode($arguments)]), 1426767159);
+        }
+        $output = explode(\PHP_EOL, ob_get_clean());
+        $count = (int)\array_pop($output);
+        if (count($output) > 0) {
+            foreach ($output as $line) {
+                $line = trim($line);
+                if ($line === '') {
+                    continue;
+                }
+                $this->outputLine('<info>+</info> %s', [$line]);
             }
         }
+        return $count < 1 ? 0 : $count;
+    }
+
+    /**
+     * Create a ContentContext based on the given workspace name
+     *
+     * @param string $workspaceName Name of the workspace to set for the context
+     * @param array $dimensions Optional list of dimensions and their values which should be set
+     * @return Context
+     */
+    protected function createContentContext($workspaceName, array $dimensions = [])
+    {
+        $contextProperties = [
+            'workspaceName' => $workspaceName,
+            'invisibleContentShown' => true,
+            'inaccessibleContentShown' => true
+        ];
+
+        if ($dimensions !== []) {
+            $contextProperties['dimensions'] = $dimensions;
+            $contextProperties['targetDimensions'] = array_map(function ($dimensionValues) {
+                return array_shift($dimensionValues);
+            }, $dimensions);
+        }
+
+        return $this->contextFactory->create($contextProperties);
     }
 
     /**
      * Create a new index with the given $postfix.
      *
      * @param string $postfix
+     * @param array $dimensionValues
      * @return void
-     * @throws CRAException
      */
-    protected function createNewIndex(string $postfix): void
+    protected function createNewIndex($postfix, array $dimensionValues = [])
     {
-        $this->nodeIndexer->setIndexNamePostfix($postfix ?: (string)time());
+        $this->nodeIndexer->setIndexNamePostfix((string)$postfix);
         if ($this->nodeIndexer->getIndex()->exists() === true) {
             $this->logger->log(sprintf('Deleted index with the same postfix (%s)!', $postfix), LOG_WARNING);
             $this->nodeIndexer->getIndex()->delete();
         }
         $this->nodeIndexer->getIndex()->create();
-        $this->logger->log('Created index ' . $this->nodeIndexer->getIndexName());
+        $this->logger->log('Created index ' . $this->nodeIndexer->getIndexName(), LOG_INFO);
+        $this->logger->log('+ Dimensions: ' . \json_encode($dimensionValues), LOG_INFO);
```

Comment:
Such log entries look really nice with the CLI logger but are kind of useless, in the log, especially when analyzed for eg with kibana, as the log message has no context at all. I would separate the logging to the console and the general logging in this case.